src/ng/compile.js

```diff
@@ -3032,6 +3053,29 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             removeWatchCollection.push(removeWatch);
             break;
 
+          case '<':
+            if (!hasOwnProperty.call(attrs, attrName)) {
+              if (optional) break;
+              attrs[attrName] = void 0;
+            }
+            if (optional && !attrs[attrName]) break;
+
+            parentGet = $parse(attrs[attrName]);
+
+            destination[scopeName] = parentGet(scope);
+
+            if (definition.collection) {
+              removeWatch = scope.$watchCollection(parentGet, function parentCollectionValueWatchAction(newParentValue) {
```

Comment:
> we are not watching the result of the array after it went thru the filter, but before

I don't think that's right. Just doing `$rootScope.$watch("a | orderBy"); $rootScope.$apply("a = []")` causes an infinite digest.  I believe `=*` was created specifically to avoid this. However I think `=*` is a bad solution, it would be better if $parse/$watch handled this somehow...